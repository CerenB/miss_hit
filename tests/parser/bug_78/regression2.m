function x = regression2
    persistent y z, if isempty(y), y = 42; end
    x = y;
end
