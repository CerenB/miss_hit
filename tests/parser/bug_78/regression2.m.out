In regression2.m, line 4
|     persistent y z, if isempty(y), y = 42; end
|                   ^ style: end statement with a semicolon instead of comma
In regression2.m, line 4
|     persistent y z, if isempty(y), y = 42; end
|                   ^ style: end statement with a newline
In regression2.m, line 4
|     persistent y z, if isempty(y), y = 42; end
|                                  ^ style: end statement with a semicolon instead of comma
In regression2.m, line 4
|     persistent y z, if isempty(y), y = 42; end
|                                  ^ style: end statement with a newline
In regression2.m, line 4
|     persistent y z, if isempty(y), y = 42; end
|                                          ^ style: end statement with a newline
----------------------------------------------------------------------
--  Parse tree for regression2.m
Root: Function_File
  Functions: Function_Definition
    Signature: Function_Signature
      Name: Identifier <regression2>
      Outputs: Identifier <x>
    Body: Sequence_Of_Statements
      Statements: Persistent_Statement
        Names: Identifier <y>
        Names: Identifier <z>
      Statements: If_Statement
        Action: If Action
          Guard: Reference
            Name: Identifier <isempty>
            Arguments: Identifier <y>
          Body: Sequence_Of_Statements
            Statements: Simple_Assignment_Statement
              LHS: Identifier <y>
              RHS: Number_Literal <42>
      Statements: Simple_Assignment_Statement
        LHS: Identifier <x>
        RHS: Identifier <y>
----------------------------------------------------------------------
MISS_HIT Summary: 1 file(s) analysed, 5 style issue(s)
